src/evaluate/evaluator.py

```diff
@@ -12,95 +12,125 @@
-            baseline_pipe is None and baseline_label_mapping is not None
-        ), "You can only provide `baseline_label_mapping` if `baseline_pipe` is provided."
-        predictions = (
-            baseline_pipe(self.data["inputs"], truncation=True)
-            if baseline_pipe is not None
-            else self.pipe(self.data["inputs"], truncation=True)
-        )
-        label_mapping = self.label_mapping if baseline_label_mapping is None else baseline_label_mapping
+        raise NotImplementedError()
+
+
+class TextClassificationEvaluator(Evaluator):
+    def __init__(self, task="text-classification", default_metric=None):
+        super().__init__(task, default_metric=default_metric)
+
+    def _compute_predictions(self, pipe: Pipeline, inputs, label_mapping: Dict[str, Number] = None) -> List[Number]:
+        predictions = pipe(inputs, truncation=True)
         return [
             label_mapping[element["label"]] if label_mapping is not None else element["label"]
             for element in predictions
         ]
 
-    def get_references(self):
-        return self.data["references"]
-
-    def statistic(self, predictions=None, references=None, metric_key: Optional[str] = None):
-        return self.metric.compute(predictions=predictions, references=references)[
-            self.metric_key if metric_key is None else metric_key
-        ]
-
-    def get_confidence_interval(self, confidence_level: float = 0.95, n_resamples: int = 9999):
-        predictions = self.get_predictions()
-        references = self.get_references()
-        return bootstrap(data=(predictions, references), statistic=self.statistic, paired=True, vectorized=False)
-
-    def get_bootstrap_p_value(
+    def compute(
         self,
-        baseline_pipe: Union[str, Pipeline],
-        baseline_label_mapping: Optional[Dict[str, Number]] = None,
+        model: str = None,
+        pipe: Union[Pipeline, Callable] = None,
+        data: Union[str, Dataset] = None,
+        metric: Union[str, Metric] = None,
+        strategy: Literal["simple", "bootstrap"] = "simple",
+        confidence_level: float = 0.95,
         n_resamples: int = 9999,
-    ):
-        """Compare two model's scores by getting p-values from non-parametric bootstrap. Here, the null hypothesis is:
-        "current model is no better than the baseline model on the test population".
-        The implementation is based on the paper: https://aclanthology.org/D12-1091.pdf.
-        """
-        baseline_pipe = pipeline(baseline_pipe) if isinstance(baseline_pipe, str) else baseline_pipe
-        preds = np.array(self.get_predictions())
-        baseline_preds = np.array(
-            self.get_predictions(baseline_pipe=baseline_pipe, baseline_label_mapping=baseline_label_mapping)
+        input_column: str = "inputs",
+        label_column: str = "references",
+        label_mapping: Optional[Dict[str, Number]] = None,
+    ) -> Tuple[Dict[str, float], Any]:
+        assert data is not None
```

Comment:
Could we add Errors instead of asserts with messages that guide the user to fix the problem?